presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -356,11 +356,11 @@ protocol::TaskStatus PrestoTask::updateStatusLocked() {
     info.taskStatus.completedDriverGroups.push_back({true, splitGroupId});
   }
 
-  const auto veloxMemStats = task->pool()->stats();
-  info.taskStatus.memoryReservationInBytes = veloxMemStats.currentBytes;
+  const auto veloxTaskMemStats = task->pool()->stats();
+  info.taskStatus.memoryReservationInBytes = veloxTaskMemStats.currentBytes;
   info.taskStatus.systemMemoryReservationInBytes = 0;
   info.taskStatus.peakNodeTotalMemoryReservationInBytes =
-      veloxMemStats.peakBytes;
+      task->queryCtx()->pool()->peakBytes();
```

Comment:
Perhaps, add a comment to explain that a given worker may run multiple tasks from the same query. peakNodeTotalMemoryReservationInBytes represents peak memory usage across all these tasks.